src/main/java/io/pravega/perf/WriterWorker.java

```diff
@@ -71,57 +99,105 @@ private String createPayload(int size) {
      */
     public abstract void flush();
 
+    /**
+     * Flush the producer data.
+     */
+    public abstract void close();
+
+
     @Override
     public Void call() throws InterruptedException, ExecutionException, IOException {
         perf.benchmark();
         return null;
     }
 
+
     private class EventsWriter implements Performance {
 
         public void benchmark() throws InterruptedException, IOException {
-            final EventsController eCnt = new EventsController(System.currentTimeMillis(), eventsPerSec);
             for (int i = 0; i < events; i++) {
                 recordWrite(payload, stats::recordTime);
-                eCnt.control(i);
             }
             flush();
         }
     }
 
-    private class EventsWriterRW implements Performance {
+
+    private class EventsWriterSleep implements Performance {
```

Comment:
Can these be made static? createBenchmark() is currently creating a loop where this holds onto the outer class which also holds it. This is not just messy but there are going to be a ton of synthetic accessors which I'd like to avoid in a benchmark.

Reply:
ok, I got your point here.
so , I have done here is , instead of creating a class which derives the Performance , i am directly exporting the methods as interface. please have look into these new changes